internal/bminventory/inventory_v2_handlers.go

```diff
@@ -605,13 +606,63 @@ func (b *bareMetalInventory) signURL(ctx context.Context, infraEnvID, urlString,
 	return urlString, &expiresAt, nil
 }
 
-const ipxeScriptFormat = `#!ipxe
+const ipxeFirstScriptFormat = `#!ipxe
+chain %s&mac=${net0/mac}
+`
+
+const ipxeSecondScriptFormat = `#!ipxe
 initrd --name initrd %s
 kernel %s initrd=initrd coreos.live.rootfs_url=%s random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8"
 boot
 `
 
-func (b *bareMetalInventory) infraEnvIPXEScript(ctx context.Context, infraEnv *common.InfraEnv) (string, error) {
+func (b *bareMetalInventory) firstIPXEScript(infraEnv *common.InfraEnv) (string, error) {
+	serviceBaseURL := b.ServiceBaseURL
+	if b.InsecureServiceBaseURL != "" {
+		serviceBaseURL = b.InsecureServiceBaseURL
+	}
+	parsedURL, err := url.Parse(serviceBaseURL)
+	if err != nil {
+		return "", err
+	}
+	builder := installer.V2DownloadInfraEnvFilesURL{
+		InfraEnvID: *infraEnv.ID,
+		FileName:   "ipxe-script",
+	}
+	return fmt.Sprintf(ipxeFirstScriptFormat, builder.StringFull(parsedURL.Scheme, parsedURL.Host)), nil
+}
+
+func (b *bareMetalInventory) shouldServeScript(infraEnv *common.InfraEnv, mac *strfmt.MAC) (bool, error) {
```

Comment:
I think this could use a better name. Maybe `shouldServeHostIPXEScript`?
That way someone immediately knows it's specific to IPXE and to an individual host.